### DISCUSSION TITLE
parse jsonlogs which contain sometimes a string instead of json

### DISCUSSION THREAD

**Discussion by guettli** [mikefarah/yq#2079] [General]
I use `yq -jp -P` to get nice yaml logs from jsonlog lines. This works fine.

But sometimes there are non-jsonlog lines:

```
{"level":"INFO","time":"2024-06-21T08:59:00.112Z","logger":"controller-runtime.certwatcher","file":"certwatcher/certwatcher.go:115","message":"Starting certificate watcher"}
I0621 08:59:00.112720      20 leaderelection.go:250] attempting to acquire leader lease caph-system/hetzner.cluster.x-k8s.io...
```

This fails:

> Error: bad file '-': invalid character 'I' looking for beginning of value

It would be great if yq could parse that somehow.

I looked at `yq --help`, but found nothing which seem helpful for that case.

What do you think about that use-case?

Do you want to support it?

Maybe I could create a PR, but feel free to do it. I guess you are way faster than me implementing it :-)

**Comment by mikefarah** [comment]
Hmm I can't think of a great way to do that, as I just pass in the multi-line JSON string into a JSON decoder, which expects JSON. Simplest thing I can think of is to filter out non-json strings (or convert them to JSON) using something like `sed` first, and then piping the result (which is now all json) into yq.
